core/src/test/java/org/elasticsearch/index/query/IndicesQueryBuilderTest.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.search.Query;
+import org.junit.Test;
+
+import java.io.IOException;
+
+public class IndicesQueryBuilderTest extends BaseQueryTestCase<IndicesQueryBuilder> {
+
+    //NO COMMIT if we want to directly compare the inner queries
+    @Override
+    protected boolean supportsBoostAndQueryName() {
```

Comment:
I don't think we need this here. very similar to discussion here: https://github.com/elastic/elasticsearch/pull/12037#discussion_r36962184 . it is needed because you use lucene equals to compare queries and because the boost from the main query overrides the inner boosts.